x-pack/plugin/spatial/src/test/java/org/elasticsearch/xpack/spatial/index/mapper/GeoShapeWithDocValuesFieldTypeTests.java

```diff
@@ -150,7 +150,61 @@ private Geometry normalize(Geometry geometry) {
         }
     }
 
-    public void testFetchSourceValueDateLine() throws IOException {
+    public void testFetchSourcePolygonDateLine() throws IOException {
+        assertFetchSourceGeometry(
+            "POLYGON((170 -10, -170 -10, -170 10, 170 10, 170 -10))",
+            "MULTIPOLYGON (((180.0 -10.0, 180.0 10.0, 170.0 10.0, 170.0 -10.0, 180.0 -10.0)),"
```

Comment:
I'm curious why the polygon test has a single polygon split to a multipolygon by the function:

    fetchSourceValue(mapper, sourceValue, "wkt")

But the BBOX is not split by the same function in the envelope test?

Perhaps this is not part of this PR, since this PR focuses on MVT, and the MVT call:

    fetchSourceValue(mapper, sourceValue, "mvt(0/0/0@4096)")

Does seem to split both the polygon and the envelope.

Reply:
I thought about the same. The didn't like the options I had to return a split geometry. Either return a multipolygon which didn't feel right for an input envelope, or a geometry collection containing two envelopes which I did not like it either.

Therefore I decided to return it unchanged. We can visit it later if it dies not feel right.